stream_alert/classifier/payload/payload_base.py

```diff
@@ -0,0 +1,222 @@
+"""
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from abc import ABCMeta, abstractmethod, abstractproperty
+import json
+import logging
+
+from stream_alert.shared.logger import get_logger
+
+
+LOGGER = get_logger(__name__)
+LOGGER_DEBUG_ENABLED = LOGGER.isEnabledFor(logging.DEBUG)
+
+
+class PayloadRecord(object):
```

Comment:
i'm trying to figure out where the line between the payload and the record is wrt deserialization. i think that's what the above line i was confused about has to do with, deserializing the wrapper payload. if that's the case, is that a parser concern, or a payload concern? if it's not the case and i'm confused then i'm sorry